Timer/clock.go

```diff
@@ -0,0 +1,74 @@
+package clock
+
+import (
+	"consumer"
+	"fmt"
+	"time"
+
+	database "github.com/pikapika/database"
+	service "github.com/pikapika/notification_services"
+)
+
+func eventToConsumerEvent(event database.Event) consumer.Event {
+	return consumer.Event{
+		Message:            event.Message,
+		Source:             event.Source,
+		DestinationAddress: event.DestinationAddress,
+		Topics:             event.Topics,
+		NotificationType:   event.NotificationType,
+		Status:             event.Status,
+		NextRetry:          event.NextRetry,
+		Attempts:           event.Attempts,
+	}
+}
+
+func RetryClock() {
+	for {
+		events, err := database.GetIncompleteEvents()
+		if err != nil {
+			fmt.Print("Error getting incomplete events", err)
+			continue
+		}
+
+		for _, event := range events {
+			if event.NextRetry > time.Now().Unix() {
+				continue
+			}
```

Comment:
is this really needed? - P0


check if you can handle it directly on query level

Reply:
Once you do this, the function `GetIncompleteEvents` name should be changed to something which ties to its name